### DISCUSSION TITLE
Is there a standard "font" for use in icons?

### DISCUSSION THREAD

**Discussion by tryoxiss** [lucide-icons/lucide#1920] [Q&A] [answered]
While most icons have no text, a good few do. For example `arrow-down-a-z`. Is there a standard "font" of SVG objects/paths for use in creating icons? Are they just copy+pasted from another icon or created as-needed?

**Comment by karsa-mistmere** [comment, marked as answer]
Unfortunately there isn't, which has already led to a lot of inconsistencies, I'm working on defining a set of letters and numerals in various sizes that could be used as a standard, but it's not quite finished yet and we'll need to take into consideration the requirements of current icons.
![image](https://github.com/lucide-icons/lucide/assets/17746067/b86a2040-d6af-4bd1-b7ca-af44d5bb9a99)

**Reply by tryoxiss** [reply]
Ahh, thats unfortunte. The letters you have there look mostly good and feel consistant with the style. A few comments though would be that the J, R, S, W and Pfeel off. My guess is its because the curve is realtively short but very long, I am not sure how I would fix this though as you probably want them to take up simillar space. As for the W it just feels sqished. /pos

**Reply by zoharma** [reply]
I would add that W and V don't need to have the same angle, and will allow the W to be more open.

Also, A is too wide, and should be like the V

**Comment by jamiemlaw** [comment]
I recently had a go at a Lucide-style alphabet, which is very similar but with a few modifications. Some of the letters have more-rounded corners, and the S is more curved.

![image](https://github.com/user-attachments/assets/d017e597-a149-4cf6-981d-17eaf38471b1)

https://codepen.io/jamiemlaw/pen/EaxOPBK

**Reply by karsa-mistmere** [reply]
Yeah, I've also done a _lot_ of work on this:
<img width="1073" alt="image" src="https://github.com/user-attachments/assets/0f5f6842-e333-4bf9-b4f5-51485236d77f" />

**Reply by jamiemlaw** [reply]
Those are some beautiful looking numerals, in particular!

**Reply by jamiemlaw** [reply]
More from my backlog:

![image](https://github.com/user-attachments/assets/fb5c05b8-a461-4541-b1a0-14def21875db)
![image](https://github.com/user-attachments/assets/05a508ce-ba72-46e4-8e8e-7788e763c386)

**Reply by karsa-mistmere** [reply]
@jamiemlaw: The double stroke serifs are super cool, not sure about the use cases yet, but I love them. 🤩

**Reply by jamiemlaw** [reply]
@karsa-mistmere It started when a large number of people were requesting a blackboard-bold letter X "for some reason"! So I thought I'd try the whole alphabet.

The P, N, Z, Q, R and C are used as symbols in mathematics to refer to certain sets of numbers. But yes, they are very niche!

**Reply by jwlinqx** [reply]
> More from my backlog:
> 
> ![image](https://private-user-images.githubusercontent.com/956081/428709029-fb5c05b8-a461-4541-b1a0-14def21875db.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODAwNzM5MDYsIm5iZiI6MTc4MDA3MzYwNiwicGF0aCI6Ii85NTYwODEvNDI4NzA5MDI5LWZiNWMwNWI4LWE0NjEtNDU0MS1iMWEwLTE0ZGVmMjE4NzVkYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyOVQxNjUzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZGE2NGVkZmFjOTJmZDc5Mjk2NmRjZGNiOTAzMmRjM2JhMzQxMmYzMzRkMDUxNmNkMzc3ZDQ0MTFjNzFkM2RlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.tihCUjavzrrJMy2Q-MB-AS0eIvA7JyMNmLqcX818j8g) ![image](https://private-user-images.githubusercontent.com/956081/428709167-05a508ce-ba72-46e4-8e8e-7788e763c386.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODAwNzM5MDYsIm5iZiI6MTc4MDA3MzYwNiwicGF0aCI6Ii85NTYwODEvNDI4NzA5MTY3LTA1YTUwOGNlLWJhNzItNDZlNC04ZThlLTc3ODhlNzYzYzM4Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyOVQxNjUzMjZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YzA4NDVkMDJlYjA2NmMwNmIzMWQ3Y2M4ZjFmZjU1OWQzZTE4MjU5ZGE4YWM3NTkwOWZmOGNmYjQwMDEwYjZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.V6uOwqlK6HR3xozpw9kTVvlbFPai_ghIBpWrVtosxq4)

I feel like your lowercase and my uppercase could be friends potentially for a san-serif font. I have also found myself needing letters or greek letters for math symbols. I work for a B to B company that has a lot of products for different calculations.

**Comment by jwlinqx** [comment]
Hey — wanted to share that I've drawn a full uppercase A–Z set against the Lucide design guide (24×24 canvas, 2px centered strokes, round caps/joins, ~3 points of precision) and would love to gauge interest before opening a PR, since this discussion is already exploring whether a standard letter set belongs in Lucide.
Why I think a letter set earns its place:

Keyboard shortcut indicators — hotkey hints in UIs, in-game prompts, "Press A to confirm" tutorials.
Avatar / user-initial placeholders — letter-shaped avatars for users without uploaded photos (common in chat, mail, project tools).
Alphabetical labels — multi-choice questions (Option A/B/C/D), diagram callouts, ordered list markers, indexed sections.
Educational / phonics / spelling apps — alphabet tracing, language learning.
Composite/acronym icons — combining individual letters to build acronym-style glyphs for product/feature icons. (This is my current use case.)

Naming proposal: letter-a-uppercase … letter-z-uppercase
I considered three options:

1. Bare letter (a, b, …) — matches the prefix pattern of a-arrow-down, but feels strange as a top-level icon name and leaves no room for lowercase without retroactive renaming.
2. letter-a — clean, but forces a rename when lowercase ships (it would have to become letter-a-uppercase or similar after the fact)
3. letter-a-uppercase — pairs cleanly with a future letter-a-lowercase set, makes case explicit (good for search/screen readers), and still follows the "depict what it is" rule (an uppercase A is literally what's depicted).

I'm leaning toward option 3 for forward compatibility, but happy to defer to whatever convention you prefer.
Authorship: Drawn from scratch, but referenced the existing Lucide letter glyphs in a-arrow-down, a-arrow-up, a-large-small, and case-upper to keep the A consistent with what's already in the library, then derived the rest of the alphabet in the same style.
Open questions before I PR:

Naming preference (above)?
All 26 in one PR, or staged in groups?
Any letters you'd want to reject upfront? I missed the case-upper icon so my "B" is a little different. 

i am attaching a zipfile of the icons but you can also see my branch with the letters: [studio/alphabet-latin](https://github.com/jwlinqx/lucide/tree/studio/alphabet-latin)

here is my figma component. far left is 8x8 frame with stroke width of 1.5px (just because its so small). then 12x12, 16x16, 20x20, and 24x24 each with 2px stroke width.
[letters-uppercase.zip](https://github.com/user-attachments/files/27610530/letters-uppercase.zip)
<img width="204" height="1502" alt="image" src="https://github.com/user-attachments/assets/6aa0b8de-1154-4c43-9bff-3e9fc54afbfe" />


i also made some numerals as well: 
<img width="200" height="704" alt="image" src="https://github.com/user-attachments/assets/f2e2faeb-91a6-43d4-9f91-ca2e4938fd38" />

**Reply by jwlinqx** [reply]
> Hey — wanted to share that I've drawn a full uppercase A–Z set against the Lucide design guide (24×24 canvas, 2px centered strokes, round caps/joins, ~3 points of precision) and would love to gauge interest before opening a PR, since this discussion is already exploring whether a standard letter set belongs in Lucide. Why I think a letter set earns its place:
> 
> Keyboard shortcut indicators — hotkey hints in UIs, in-game prompts, "Press A to confirm" tutorials. Avatar / user-initial placeholders — letter-shaped avatars for users without uploaded photos (common in chat, mail, project tools). Alphabetical labels — multi-choice questions (Option A/B/C/D), diagram callouts, ordered list markers, indexed sections. Educational / phonics / spelling apps — alphabet tracing, language learning. Composite/acronym icons — combining individual letters to build acronym-style glyphs for product/feature icons. (This is my current use case.)
> 
> Naming proposal: letter-a-uppercase … letter-z-uppercase I considered three options:
> 
> 1. Bare letter (a, b, …) — matches the prefix pattern of a-arrow-down, but feels strange as a top-level icon name and leaves no room for lowercase without retroactive renaming.
> 2. letter-a — clean, but forces a rename when lowercase ships (it would have to become letter-a-uppercase or similar after the fact)
> 3. letter-a-uppercase — pairs cleanly with a future letter-a-lowercase set, makes case explicit (good for search/screen readers), and still follows the "depict what it is" rule (an uppercase A is literally what's depicted).
> 
> I'm leaning toward option 3 for forward compatibility, but happy to defer to whatever convention you prefer. Authorship: Drawn from scratch, but referenced the existing Lucide letter glyphs in a-arrow-down, a-arrow-up, a-large-small, and case-upper to keep the A consistent with what's already in the library, then derived the rest of the alphabet in the same style. Open questions before I PR:
> 
> Naming preference (above)? All 26 in one PR, or staged in groups? Any letters you'd want to reject upfront? I missed the case-upper icon so my "B" is a little different.
> 
> i am attaching a zipfile of the icons but you can also see my branch with the letters: [studio/alphabet-latin](https://github.com/jwlinqx/lucide/tree/studio/alphabet-latin)
> 
> here is my figma component. far left is 8x8 frame with stroke width of 1.5px (just because its so small). then 12x12, 16x16, 20x20, and 24x24 each with 2px stroke width. [letters-uppercase.zip](https://github.com/user-attachments/files/27610530/letters-uppercase.zip) <img alt="image" width="204" height="1502" src="https://private-user-images.githubusercontent.com/245216054/590687318-6aa0b8de-1154-4c43-9bff-3e9fc54afbfe.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODAwNzM5MjcsIm5iZiI6MTc4MDA3MzYyNywicGF0aCI6Ii8yNDUyMTYwNTQvNTkwNjg3MzE4LTZhYTBiOGRlLTExNTQtNGM0My05YmZmLTNlOWZjNTRhZmJmZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyOVQxNjUzNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iODA3ZDA3NzEwMjEzOTExOTY4ZDZjMGZmMzlkNTY4NWVjYjcyMTA4ZDU3ZmFhZDU2NWE1NGU5YmViZGRjZGNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.XqcyL2UyTiOyq1K5q5UXLzQ0afwdOnzeJRKdynoG26U">
> 
> i also made some numerals as well: <img alt="image" width="200" height="704" src="https://private-user-images.githubusercontent.com/245216054/590687975-f2e2faeb-91a6-43d4-9f91-ca2e4938fd38.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3ODAwNzM5MjcsIm5iZiI6MTc4MDA3MzYyNywicGF0aCI6Ii8yNDUyMTYwNTQvNTkwNjg3OTc1LWYyZTJmYWViLTkxYTYtNDNkNC05ZjkxLWNhMmU0OTM4ZmQzOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDUyOVQxNjUzNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNTYyYTlkODkzOTRjMjJmZDMwYzZhNWI2NmNmZGY5MDhhY2UxMzA2ZTc4YmRjMmEwOTA1NWRiOGY0YTdhODZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.SVMDrOcJwI1Th2E7AlqFmjnMcVl1Ei7E9IV--eUX8Uo">

I updated some of the letters so that zip file will be off. the branch has the new ones though
